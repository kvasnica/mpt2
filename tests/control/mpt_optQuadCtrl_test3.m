function mpt_optQuadCtrl_test3

% test CFTOC for PWA systems with quadratic cost
% focus on boolean/integer inputs

% 2 states, one output, yref given, 1st input real, 2nd input boolean
two_tanks
probStruct.N = 2; probStruct.norm = 2; probStruct.Tconstraint=0;
ctrl = mpt_control(sysStruct, probStruct);


%mbg_assertequal(length(ctrl), 19);
mbg_asserttrue(length(ctrl) >= 19);

% verify that we reach a given reference
[x, u, y] = sim(ctrl, [0.5; 0.17], 20);
mbg_asserttolequal(y(end), probStruct.yref, 1e-2);



% 2 states, one output, yref given, 1st input integer, 2nd input boolean
two_tanks
sysStruct.Uset{1} = [0 0.4 0.6];
probStruct.N = 1; probStruct.norm = 2; probStruct.Tconstraint=0;
ctrl = mpt_control(sysStruct, probStruct);

%mbg_assertequal(length(ctrl), 4);
mbg_asserttrue(length(ctrl) > 2);
