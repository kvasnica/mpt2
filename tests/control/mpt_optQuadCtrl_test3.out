	
No terminal weight "probStruct.P_N" specified... closed-loop stability will not be guaranteed
mpt_mpqp: 1 regions
mpt_mpqp: 1 regions
mpt_mpqp: Infeasible optimization problem from the begining
mpt_mpqp: 2 regions
mpt_mpqp: 1 regions
mpt_mpqp: 1 regions
mpt_mpqp: Infeasible optimization problem from the begining
mpt_mpqp: 2 regions
mpt_mpqp: 2 regions
mpt_mpqp: 1 regions
mpt_mpqp: 2 regions
mpt_mpqp: 3 regions
mpt_mpqp: 2 regions
mpt_mpqp: 1 regions
mpt_mpqp: 2 regions
mpt_mpqp: 3 regions
-> Generated 9 partitions.
Solution consists of 19 regions
No terminal weight "probStruct.P_N" specified... closed-loop stability will not be guaranteed
-> Generated 4 partitions.
Solution consists of 4 regions